<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- args to be passed in -->
  <arg name="frame_rate" value="30" />
  <arg name="show" default="true" />
  <arg name="calibrated" default="true" />

  <node name="pointgrey_camera" pkg="pointgrey_camera_driver" type="camera_node">

  	<param name="camera_info_url" if="$(arg calibrated)"
        value="file:///home/jesse/aruco_test_ws/src/params/chameleon3_LLNL.yaml" />
    <param name="frame_rate" value="$(arg frame_rate)" />
    <param name="auto_exposure" value="false" />
    <param name="exposure" value="-1.4" />
    <param name="auto_shutter" value="false" />
    <param name="shutter_speed" value="0.01" />
    <param name="format7_color_coding" value="mono8" />
    <param name="auto_white_balance" value="false" />
    <param name="auto_gain" value="false" />
    <param name="gain" value="10.0" />
    <param name="brightness" value="1.4" />
    <param name="gamma" value="2.0" />
  </node>

  <node pkg="aruco_localization" type="aruco_localization" name="aruco">
    <param name="show_output_video" value="$(arg show)" />
    <param name="markermap_config" value="/home/jesse/aruco_test_ws/src/params/marker_166.yaml" />
    <param name="marker_size" value="0.131" />

    <param name="debug_save_input_frames" value="false" />
    <param name="debug_save_output_frames" value="false" />
    <param name="debug_image_path" value="/tmp/arucoimages" />

    <remap from="input_image" to="image_raw" />
    <remap from="output_image" to="aruco/image" />
  </node>

  <!-- node for running image_intensity_info_publisher -->
  <node pkg="misc_pkg" type="image_intensity_info_publisher.py" name="intensity_info_publisher">
    <param name="intensity_threshold" value="50.0" />
  </node>

  <group if="$(arg show)">
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
      <remap from="image" to="image_raw"/>
      <param name="autosize" value="true" />
    </node>
  </group>
</launch>
